src/xb-node.c

```diff
@@ -23,6 +23,30 @@ typedef struct {
 G_DEFINE_TYPE_WITH_PRIVATE (XbNode, xb_node, G_TYPE_OBJECT)
 #define GET_PRIVATE(o) (xb_node_get_instance_private (o))
 
+/**
+ * XbNodeIter:
+ *
+ * A XbNodeIter structure represents an iterator that can be used
+ * to iterate over the attributes of a #XbNode. XbNodeIter
+ * structures are typically allocated on the stack and then initialized
+ * with %xb_node_attr_iter_init().
+ *
+ * The iteration order of a #XbNodeIter is not defined.
+ */
+
+typedef struct
+{
+  XbNode 	*node;
+  guint8	position;
+  gpointer	dummy3;
+  gpointer	dummy4;
+  gpointer	dummy5;
+  gpointer	dummy6;
+} RealIter;
```

Comment:
Might want to call this `RealNodeIter` in case another kind of iter is added to this file in future.